stream/src/main/java/org/apache/kafka/streaming/processor/internals/StreamThread.java

```diff
@@ -148,11 +148,30 @@ public void close() {
 
     private void shutdown() {
         log.info("Shutting down a stream thread");
-        commitAll(time.milliseconds());
 
-        producer.close();
-        consumer.close();
-        removePartitions();
+        // Exceptions should not prevent this call from going through all shutdown steps.
+
+        try {
+            commitAll(time.milliseconds());
```

Comment:
commitAll() is also called from onPartitionsRevoked(). I am not sure how KafkaConsumer handles a runtime exception from ConsumerRebalanceCallback.